supporter-product-data/src/main/scala/com/gu/services/SelectActiveRatePlansQuery.scala

```diff
@@ -0,0 +1,39 @@
+package com.gu.services
+
+import com.gu.model.FieldsToExport._
+import com.gu.model.Stage
+import com.gu.model.Stage.{DEV, PROD}
+
+import java.time.LocalDate
+
+object SelectActiveRatePlansQuery {
+
+  val name = "select-active-rate-plans"
+
+  val isNotDSGift = "Subscription.RedemptionCode__c = ''"
+  val isRedeemedDSGift = s"(Subscription.RedemptionCode__c != '' AND ${gifteeIdentityId.zuoraName} != '')"
+
+  def excludeDiscountProductRatePlans(discountProductRatePlanIds: List[String]) =
+    discountProductRatePlanIds
+      .map(discountProductRatePlanId => s"${productRatePlanId.zuoraName} != '$discountProductRatePlanId' AND")
+      .mkString("\n")
+
+  def query(date: LocalDate, discountProductRatePlanIds: List[String]): String =
+    s"""SELECT
+          ${identityId.zuoraName},
+          ${gifteeIdentityId.zuoraName},
+          ${ratePlanId.zuoraName},
+          ${productRatePlanId.zuoraName},
+          ${productRatePlanName.zuoraName},
+          ${termEndDate.zuoraName}
+            FROM
+            rateplan
+            WHERE
+            ${termEndDate.zuoraName} >= '$date' AND
```

Comment:
that's good, so even if we change the date, it still carries on from where we left off?

Reply:
oh ignore that, I realised it doesn't submit any query on subsequent increments